File: Pesquisa/java/Pesquisa/src/pesquisa/Pesquisa.java
package pesquisa;

import java.util.ArrayList;

public class Pesquisa {
	public boolean pesquisaSequencial(ArrayList<Integer> lista, int valor) {
		for(int i = 0; i < lista.size(); i++) {
			if(valor == lista.get(i)) {
				return true;
			}
			if(valor < lista.get(i)) {
				break;
			}
		}
		return false;
	}
	
	public boolean pesquisaBinaria(ArrayList<Integer> lista, int valor) {
		int ini = 0;
		int fim = lista.size()-1;
		int meio;
		do {
			meio = (ini+fim)/2;
			if(valor == lista.get(meio)) {
				return true;
			}
			if(valor < lista.get(meio)) {
				fim = meio - 1;
			}else {
				ini = meio + 1;
			}
		}while(ini>=fim);
		return false;
	}
}

File: Pesquisa/python/pesquisa.py
class Pesquisa:
    def pesquisaSequencial(lista, valor):
        i = 0
        while(i < len(lista)):
            if(valor == lista[i]):
                return True
            if(valor < lista[i]):
                break
            i=i+1
        return False
	
    def pesquisaBinaria(lista, valor):
        ini = 0
        fim = len(lista) - 1
        while(ini >= fim):
            meio = (ini+fim)/2
            if(valor == lista[meio]):
                return True
 			
                if(valor < lista[meio]):
                    fim = meio - 1
                else:
                    ini = meio + 1
                
        return False